### PR TITLE
Add support for .NET nanoFramework

### LINKED ISSUE #294: port for .NET nanoFramework
Hi,

Since .NETMF become defunct, I've been working along with other folks on nanoFramework, a revised and augment spinoff of .NETMF.
We now have networking capabilities and therefore having AMQP component is a must have. 
Can I submit a PR to add support for this platform?

### PR DESCRIPTION
**Pull request by josesimoes**
- Add projects for libs (standard and micro versions)
- Add test project
- Add example projects
- Add nuspecs
- Update build queue to include nF projects
- Update docs

- Fixes #294

### CODE REVIEW

**Comment by josesimoes** [comment]
@xinchen10 we have a issue with the build: AppVeyor is using VS 2015, but nanoFramework extension requires VS 2017...

**Comment by xinchen10** [comment]
I am moving the project to VS 2017 and dotnet 2.0, but the progress is slow. I don't have a committed date for completion. It is also a challenge to have netmf/nano in the same build system and nuget package as other .Net platforms. Will it work to have a separate repo for nano? We can submodule the amqp code but have nano specific build and packaging.

**Comment by josesimoes** [comment]
Yes. .NETMF is stuck on VS2015 and won't change...
What we could do is to add a second AppVeyor job on which the build for nF using VS2017 would take place. That won't force you to change anything on the current build stuff.

What do you think?

**Comment by xinchen10** [comment]
Sounds good to me.

**Comment by josesimoes** [comment]
I've tweak AppVeyor yaml:
  + added a build matrix with one job running VS2015 image an the existing build
  + added a second job to the matrix running a VS2017 image and doing the nanoFramework part of the build

Also cloned the main solution and edit the new one for VS2017 removing all the projects, except the nanoFramework related.
Left the bulk of the structure there to be picked up when moving stuff to VS2017. If that's not interesting just remove the folders.

**Comment by josesimoes** [comment]
I've just pushed a change updating the NuGets to the latest versions on nanoFramework.

Could you please let me know if you are planning to merge this or not? If not, we'll be releasing it from our repo.

Thanks!

**Comment by josesimoes** [comment]
I've updated the NuGets of nanoFramework with the latest versions (in the mean time, we've released our v1.0).

Would like to ask if you are looking into merging this. If not, we will do it from our repo.

File: netmf/NetMF/SRAmqp.cs [hunk 1]
```diff
@@ -21,112 +21,156 @@ internal partial class SRAmqp
     {
         public static string AmqpHandleExceeded
         {
+#if (MF_FRAMEWORK_VERSION_V4_2 || MF_FRAMEWORK_VERSION_V4_3 || MF_FRAMEWORK_VERSION_V4_4 || NANOFRAMEWORK_V1_0)
```
Comment (xinchen10) [inline netmf/NetMF/SRAmqp.cs]: Is this required? This file should be only included for netmf and nano frameworks.

Comment (josesimoes) [reply netmf/NetMF/SRAmqp.cs]: You have point it doesn't. Just remove all that and pushed the change.

**Comment by josesimoes** [comment]
Thank you! 👍 😃

**Pull request merged**